src/main/java/com/cloudbees/plugins/credentials/CredentialsResolver.java

```diff
@@ -126,14 +128,14 @@ protected CredentialsResolver(@NonNull Class<F> fromClass) {
             // if the reflective instantiation proves a hot point, put a cache in front.
             try {
                 @SuppressWarnings("unchecked")
-                final CredentialsResolver<Credentials, C> resolver = resolveWith.value().newInstance();
+                final CredentialsResolver<Credentials, C> resolver = resolveWith.value().getDeclaredConstructor().newInstance();
```

Comment:
Should this become a `getConstructor()` too?

Reply:
My advice is to just revert the modifications related to newInstance/constructors and I will be able to merge your PR :)

Reply:
I'm fine with that too 👍; fixup commit is on the way ...